### ISSUE TITLE
Suggestion, Seeking Suggestion, And A Question

### ISSUE THREAD

**Issue by GitHubUser5672** [UnamSanctam/UnamWebPanel#259]
Would be great if there was an option to completely remove a miner (just one entry, not both CPU/GPU), as some computers have an unsupported CPU or GPU which then stays in the list taking up space, as well as to rename added computers.

Also, which coin would you suggest for an RX 590 on 2miners, because I've set up XMR mining and it says unsupported GPU in the checker.exe (I can only change config through the web panel btw).

And last: can the config be applied while the miner is offline or does it have to be running?

**Comment by UnamSanctam** [comment]
>Would be great if there was an option to completely remove a miner (just one entry, not both CPU/GPU), as some computers have an unsupported CPU or GPU which then stays in the list taking up space, as well as to rename added computers.

You mean blacklisting it? It's technically possible though it's a bit involved since we'll need a blacklist page and such and would need to check when the miner connects (or when displaying the miner), but it is possible of course.

>Also, which coin would you suggest for an RX 590 on 2miners, because I've set up XMR mining and it says unsupported GPU in the checker.exe (I can only change config through the web panel btw).

The RX 590 should be supported, though since the checker uses the WMI (compared to the miner which reads it from the registry) it might say unsupported if the GPU does not list AMD anywhere within it's name or model (some AMD cards don't for some reason). The RX 590 should be able to mine any coin really, so mining something like ETC would work just fine. If you add an ETC miner into it and run the checker does the ETC miner correctly start? In which case the GPU is supported, if it doesn't then your RX 590 might not even say that it's AMD in the registry, which would be troubling.

>And last: can the config be applied while the miner is offline or does it have to be running?

No the miner can't retrieve the configuration while it's not running.

**Comment by UnamSanctam** [comment]
>Yes, that.

I will look into adding something like that then.

>What I meant to ask is if it will load the new configuration when it comes back online, or does it have to be online at the time of changing it?

It will load the new configuration when it comes online.

>Since I set it to mine XMR when building, I'm guessing the config only works with XMRig, therefore I can't mine coins such as ETC, right?

Yes if you compiled it with only an XMR miner then no ETC miner exists inside the program.

>Also have another suggestion: the "Google/Libs" folder that gets created is pretty suspicious lol, would be better if the folder's name was the GPU manufacturer's folder instead, and if the "ddb64.dll" file description wasn't "XMRig CUDA plugin".

The Google/Libs folder gets generated for CPU mining as well, since that's where the MSR register driver goes, it used to be called Microsoft/Libs or Microsoft/Telemetry but some game anti-cheats (and some antivirus settings) have started blocking those folder names which is why it had to be changed, and Google was chosen since nearly everyone has something installed related to Google. The GPU manufacturers name shouldn't really be used since they might not have one. For the `ddb64.dll` description I guess I have to modify the assembly, seems like it got reverted the last time I updated the CUDA plugin, 99% of people don't do XMR mining (therefore it's not my primary focus) so I didn't notice it.

**Comment by GitHubUser5672** [comment]
GPU mining did not work. History shows just 0, and it's the same for all of the 4 PCs I installed it to which all have the same RX 590.

Idk if it means anything but they show up as "Radeon RX 590 Series" in the web panel.


Btw, which coin requires the least specs to mine that also uses XMRig, just to see if it's an issue with the miner or something else.


And I noticed green and orange colors in the program depending on which coin I choose, I know orange is XMRig and idk what the green one is.

**Comment by GitHubUser5672** [comment]
> What are you trying to mine with the GPU at the moment?

XMR. Works great on my own PC, giving me 1.5-2KH/s without affecting performance and makes no noise.


Also, do you think it is more efficient to run the CPU (idle) at 90% or 100%?

**Comment by UnamSanctam** [comment]
>XMR. Works great on my own PC, giving me 1.5-2KH/s without affecting performance and makes no noise.

XMR is designed on purpose to be bad at GPU mining, so usually one would opt to mine something else. The biggest problem with GPU mining XMR is usually the huge memory usage. Try to start an ETC miner on the Radeon RX 590 GPUs and see if it starts, just to see if it can recognize the GPU.

>Also, do you think it is more efficient to run the CPU (idle) at 90% or 100%?

Probably at 90%.

>And does the miner use huge pages?

Yes.

**Comment by GitHubUser5672** [comment]
> Try to start an ETC miner

Unable to do anything outside of the web panel, currently looking for some low spec coin to mine

Another question lol, does updating a configuration automatically change it for all the miners that use it?

**Comment by UnamSanctam** [comment]
>Another question lol, does updating a configuration automatically change it for all the miners that use it?

Yes, if you change a configuration then any miner that is set to use that configuration will change.

**Comment by GitHubUser5672** [comment]
Since there is no option for neoxa idk how to set it up, so is this good?
```
{
    "algo": "rx/0",
    "pool": "neox.2miners.com",
    "port": 4040,
    "wallet": "X",
    "password": "",
    "nicehash": false,
    "ssltls": false,
    "max-cpu": X,
    "idle-wait": X,
    "idle-cpu": X,
    "stealth-targets": "X",
    "kill-targets": "",
    "stealth-fullscreen": false
}
```

**Comment by GitHubUser5672** [comment]
Tried mining conceal, and it works. So looks like it's just too weak for XMR.

**Comment by GitHubUser5672** [comment]
I am glad XMR did not work, because I just found out I can get 15-20x more $ mining NEOX, it does not work on my nvidia GPU but works for the old AMD.

**Comment by UnamSanctam** [comment]
Alright that's great, XMR isn't really meant for GPU mining (on purpose) so it does have some weird requirements and quirks when doing so.
